File: Python/Easy/graphs/flood_fill.py
class Solution:
    def floodFill(image, sr, sc, color):
        m = len(image)
        n = len(image[0])
        start = image[sr][sc]

        def DFSsearch(i,j):
            if i < 0 or j < 0 or i >=m or j >= n or image[i][j] == color :
                return
            
            image[i][j] = color

            if image[i+1][j] == image[i][j]:
                DFSsearch(i+1,j)
            if image[i-1][j] == image[i][j]:
                DFSsearch(i-1,j)
            if image[i][j-1] == image[i][j]:
                DFSsearch(i, j-1)
            if image[i][j+1] == image[i][j]:
                DFSsearch(i, j+1)

        DFSsearch(sr, sc)

        return image

File: JavaScript/Easy/graphs/flood_fill.js
var floodFill = function(image, sr, sc, color) {

    if (!image || !image.length || image[sr][sc]===color) {return image}

    let m = image.length
    let n = image[0].length
    let val = image[sr][sc]

    var DFS = (i,j) => {
        if (i<0 || j <0 || i>=m || j >=n || val !== image[i][j])  {return}

        image[i][j]=color
        DFS(i+1,j)
        DFS(i-1, j)
        DFS(i,j+1)
        DFS(i,j-1)
    }

    DFS(sr,sc)
    return image
}